src/pngwriter.cc

```diff
@@ -1065,19 +964,23 @@ void pngwriter::close()
    /* key is a 1-79 character description of type char* */
    char key_title[] = "Title";
    text_ptr[0].key = key_title;
-   text_ptr[0].text = texttitle_;
+   std::vector<char> texttitle(texttitle_.begin(), texttitle_.end());
+   text_ptr[0].text = &texttitle[0];
    text_ptr[0].compression = PNG_TEXT_COMPRESSION_NONE;
    char key_author[] = "Author";
    text_ptr[1].key = key_author;
-   text_ptr[1].text = textauthor_;
+   std::vector<char> textauthor(textauthor_.begin(), textauthor_.end());
+   text_ptr[1].text = &textauthor[0];
```

Comment:
`c_str()` not points always to the internal buffer (only in C++11). 
`c_str()` can be a copy of the internal data, in that case the pointer can be invalid after the expression is evaluated.

[c_str() documentation](http://www.cplusplus.com/reference/string/string/data/)
`The pointer returned may be invalidated by further calls to other member functions that modify the object.`

To avoid side effects I wrote it save (explicit copy) but I miss to pass the `/0` to the end of the array (`textauthor.push_back('\0');`).